shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -655,96 +668,227 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
+  TRACE_EVENT0("flutter", "FlutterPlatformViewsController::CreateMissingLayers");
+
+  auto missing_layer_count = required_overlay_layers - layer_pool_->size();
+  // Workaround for FLutterPlatformViewsTest
+  if ([[NSThread currentThread] isMainThread]) {
+    // Create Missing Layers
+    for (auto i = 0u; i < missing_layer_count; i++) {
+      CreateLayer(gr_context,                                      //
+                  ios_context,                                     //
+                  ((FlutterView*)flutter_view_.get()).pixelFormat  //
+      );
+    }
+    return;
+  }
+
+  auto latch = std::make_shared<fml::CountDownLatch>(1u);
+  platform_task_runner_->PostTask([&]() {
+    // Create Missing Layers
+    for (auto i = 0u; i < missing_layer_count; i++) {
+      CreateLayer(gr_context,                                      //
+                  ios_context,                                     //
+                  ((FlutterView*)flutter_view_.get()).pixelFormat  //
+      );
+    }
+    latch->CountDown();
+  });
+  latch->Wait();
+}
+
+/// Update the buffers and mutate the platform views in CATransaction on the platform thread.
+void FlutterPlatformViewsController::PerformSubmit(
+    const LayersMap& platform_view_layers,
+    const std::vector<SurfaceFrame::DeferredSubmit>& callbacks,
+    std::map<int64_t, EmbeddedViewParams>& current_composition_params,
+    const std::unordered_set<int64_t>& views_to_recomposite,
+    const std::vector<int64_t>& composition_order,
+    const std::vector<std::shared_ptr<FlutterPlatformViewLayer>>& unused_layers,
+    const std::vector<UIView*>& views_to_dispose) {
+  TRACE_EVENT0("flutter", "FlutterPlatformViewsController::SubmitFrame::CATransaction");
+
+  [CATransaction begin];
+
+  // Configure Flutter overlay views.
+  for (const auto& [key, layers] : platform_view_layers) {
+    for (const auto& layer_data : layers) {
+      layer_data.layer->UpdateViewState(flutter_view_,         //
+                                        layer_data.rect,       //
+                                        layer_data.view_id,    //
+                                        layer_data.overlay_id  //
+      );
+    }
+  }
+
+  // Dispose unused Flutter Views.
+  for (auto& view : views_to_dispose) {
+    [view removeFromSuperview];
+  }
+
+  // Composite Platform Views.
+  for (int64_t view_id : views_to_recomposite) {
+    CompositeWithParams(view_id, current_composition_params[view_id]);
+  }
+
+  // Present callbacks.
+  for (const auto& cb : callbacks) {
+    cb();
+  }
+
+  // Organize the layers by their z indexes.
+  auto active_composition_order = BringLayersIntoView(platform_view_layers, composition_order);
+
+  // If a layer was allocated in the previous frame, but it's not used in the current frame,
+  // then it can be removed from the scene.
+  RemoveUnusedLayers(unused_layers, composition_order, active_composition_order);
 
   // If the frame is submitted with embedded platform views,
```

Comment:
Isn't this comment outdated now?

Reply:
Yep -- I'd be tempted to just delete the whole comment.